sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -240,9 +240,14 @@ static bool isDeviceBinaryTypeSupported(const context &C,
   }
 
   // OpenCL 2.1 and greater require clCreateProgramWithIL
-  if ((ContextBackend == backend::opencl) &&
-      C.get_platform().get_info<info::platform::version>() >= "2.1")
-    return true;
+  if (ContextBackend == backend::opencl) {
+    // According to the spec, the version string has the following format:
+    // OpenCL<space><major_version.minor_version><space><platform-specific info>
+    auto VersionStringVector =
+        split_string(C.get_platform().get_info<info::platform::version>(), ' ');
+    if ((VersionStringVector.size() > 1) && (VersionStringVector[1] >= "2.1"))
```

Comment:
Lexicographical comparison is unreliable here. I guess "12.1" >= "2.1" would evaluate to false.